Library/Formula/bash-completion.rb

```diff
@@ -32,8 +33,15 @@ def install
 
     system "./configure", "--prefix=#{prefix}"
     system "make", "install"
+  end
 
-    unless (compdir/"brew_bash_completion.sh").exist?
+  def post_install
+    # This symlink needs to be created in the post-install step. Otherwise, a
+    # copy of our official bash completion for `brew` will end up in the bottle,
+    # which is not very helpful in keeping this up-to-date as Homebrew updates.
+    brew_comp = compdir/"brew_bash_completion.sh"
+    unless brew_comp.symlink?
```

Comment:
> Perhaps for "repair requires delete" there should be some kind of lost-and-found to which the file is moved rather than deleted.

This is a very good suggestion, but sadly not available right now. I updated the PR with an expanded comment that tries to better explain what is going on and why the formula's `post_install` looks the way it does.